debezium-server/debezium-server-redis/src/test/java/io/debezium/server/redis/RedisDatabaseHistoryIT.java

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright Debezium Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.debezium.server.redis;
+
+import java.time.Duration;
+import java.util.List;
+
+import javax.enterprise.event.Observes;
+
+import org.awaitility.Awaitility;
+import org.fest.assertions.Assertions;
+import org.junit.jupiter.api.Test;
+
+import io.debezium.config.Configuration;
+import io.debezium.connector.mysql.MySqlConnection;
+import io.debezium.connector.mysql.MySqlConnection.MySqlConnectionConfiguration;
+import io.debezium.doc.FixFor;
+import io.debezium.server.TestConfigSource;
+import io.debezium.server.events.ConnectorStartedEvent;
+import io.debezium.testing.testcontainers.MySqlTestResourceLifecycleManager;
+import io.debezium.util.Testing;
+import io.quarkus.test.common.QuarkusTestResource;
+import io.quarkus.test.junit.QuarkusIntegrationTest;
+import io.quarkus.test.junit.TestProfile;
+
+import redis.clients.jedis.HostAndPort;
+import redis.clients.jedis.Jedis;
+import redis.clients.jedis.StreamEntryID;
+import redis.clients.jedis.resps.StreamEntry;
+
+/**
+ * Integration test that verifies reading and writing database history from Redis key value store
+ *
+ * @author Oren Elias
+ */
+@QuarkusIntegrationTest
+@TestProfile(RedisDatabaseHistoryTestProfile.class)
+@QuarkusTestResource(RedisTestResourceLifecycleManager.class)
+
+public class RedisDatabaseHistoryIT {
+
+    void setupDependencies(@Observes ConnectorStartedEvent event) {
+        Testing.Print.enable();
+    }
+
+    private static final String STREAM_NAME = "metadata:debezium:db_history";
+
+    protected static Jedis jedis;
+
+    @Test
+    @FixFor("DBZ-4771")
+    public void testDatabaseHistory() throws Exception {
+        jedis = new Jedis(HostAndPort.from(RedisTestResourceLifecycleManager.getRedisContainerAddress()));
+        Testing.Print.enable();
+        Awaitility.await().atMost(Duration.ofSeconds(TestConfigSource.waitForSeconds())).until(() -> {
+            final long streamLength = jedis.xlen(STREAM_NAME);
+            return streamLength > 0;
+        });
+
+        final long streamLength = jedis.xlen(STREAM_NAME);
+        Assertions.assertThat(streamLength > 0).isTrue();
```

Comment:
Why is this expected to be > 0? I.e. where's that record coming from?

Reply:
The schema of the MySQL example DB. I can also check that the actual DDL that is stored there matches the "CREATE TABLE" (assuming that order is preserved?). Should I add this or another assertion to validate the entries written?

Reply:
Yes, I think some assertion of the history contents would be warranted.

Reply:
done